debezium-core/src/main/java/io/debezium/relational/TableId.java

```diff
@@ -37,6 +37,12 @@ public static TableId parse(String str) {
     public static TableId parse(String str, boolean useCatalogBeforeSchema) {
         String[] parts = str.split("[\\" + '.' + "]");
         if ( parts.length < 0 ) return null;
+        for (int i = 0; i < parts.length; i++) {
+            String part = parts[i];
+            if ( part.startsWith("\"") && part.endsWith("\"") ) {
+                parts[i] = part.substring(1, part.length()-1);
+            }
+        }
```

Comment:
Strictly speaking, this may not be correct for all of the DBMS dialects. For example, MySQL supports other quote characters. I'd suggest adding an overloaded `parse` method that takes a string with valid quote characters, and if non-null then check for matching quote characters.

The existing `parse(String str, boolean useCatalogBeforeSchema)` method could then delegate to `parse(str, useCatalogBeforeSchema, null)`.